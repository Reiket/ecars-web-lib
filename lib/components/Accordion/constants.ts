export const ACCORDION_TEST_ID = 'accordionTestId';
