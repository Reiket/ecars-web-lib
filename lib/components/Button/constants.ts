import type {ButtonHTMLAttributes} from 'react';

export const BUTTON_SIZE = Object.freeze({
  BIG: 'big',
  SMALL: 'small',
});

export const BUTTON_COLOR = Object.freeze({
  GREEN: 'green',
  WHITE: 'white',
  GRAY: 'gray',
});


export type ButtonAttributesType = ButtonHTMLAttributes<HTMLButtonElement>['type']
export type ButtonSizeType = (typeof BUTTON_SIZE)[keyof typeof BUTTON_SIZE];
export type ButtonColorType = (typeof BUTTON_COLOR)[keyof typeof BUTTON_COLOR];

export const BUTTON_TEST_ID = 'buttonTestId';