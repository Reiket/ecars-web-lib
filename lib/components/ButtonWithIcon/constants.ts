export const LEFT_ICON_BUTTON_TEST_ID = 'leftIconButtonTestId';
export const RIGHT_ICON_BUTTON_TEST_ID = 'rightIconButtonTestId';
