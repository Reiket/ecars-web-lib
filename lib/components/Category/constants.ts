export const CATEGORIES_LIST = Object.freeze({
  GUIDES: 'guides',
  REVIEWS: 'reviews',
  NEWS: 'news',
});

export type CategoriesListType = (typeof CATEGORIES_LIST)[keyof typeof CATEGORIES_LIST];

export const CATEGORY_TEST_ID = 'categoryTestId';
