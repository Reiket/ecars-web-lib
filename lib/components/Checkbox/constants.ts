import type {ElementProps} from '@/services/types';

export interface CheckboxProps extends ElementProps {
  name?: string;
  id?: string;
  disabled?: boolean;
  onChange?: (checked: boolean) => void;
  checked?: boolean;
  hasError?: boolean;
}

export const CHECKBOX_TEST_ID = 'checkboxTestId';
export const CHECKBOX_TEST_INPUT_ID = 'checkboxTestInputId';
