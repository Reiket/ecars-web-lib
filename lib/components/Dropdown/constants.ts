import type {ReactNode} from 'react';

export interface DropdownOption {
  value: string;
  label?: ReactNode;
}

export interface DropdownProps {
  onSelect: (value: string) => void;
  category: string;
  options: DropdownOption[];
}

export const DROPDOWN_TEST_ID = 'dropdownTestId';

export const DROPDOWN_THEME = Object.freeze({
  DARK: 'dark',
  LIGHT: 'light',
});

export type DropdownThemeType = (typeof DROPDOWN_THEME)[keyof typeof DROPDOWN_THEME];
