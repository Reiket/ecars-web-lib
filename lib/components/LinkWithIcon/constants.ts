export const LEFT_ICON_LINK_TEST_ID = 'leftIconLinkTestId';
export const RIGHT_ICON_LINK_TEST_ID = 'rightIconLinkTestId';