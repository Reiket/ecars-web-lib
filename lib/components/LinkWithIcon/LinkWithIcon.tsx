import type {ComponentType, FC, ReactNode} from 'react';
import {RouterLink, RouterLinkProps} from '@/components/RouterLink/RouterLink';
import {LEFT_ICON_LINK_TEST_ID, RIGHT_ICON_LINK_TEST_ID} from '@/components/LinkWithIcon/constants';

interface Props extends RouterLinkProps {
  children: ReactNode;
  RightIcon?: ComponentType;
  LeftIcon?: ComponentType;
}

export const LinkWithIcon: FC<Props> = ({RightIcon, LeftIcon, children, ...props}) => {
  return (
    <RouterLink
      withIcon
      {...props}
    >
      {!!LeftIcon && <LeftIcon data-testid={LEFT_ICON_LINK_TEST_ID} />}
      {children}
      {!!RightIcon && <RightIcon data-testid={RIGHT_ICON_LINK_TEST_ID} />}
    </RouterLink>
  );
};
