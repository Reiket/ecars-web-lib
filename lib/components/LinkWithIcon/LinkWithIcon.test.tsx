import {cleanup, render, screen} from '@testing-library/react';
import '@testing-library/jest-dom';
import {LinkWithIcon} from '@/components/LinkWithIcon/LinkWithIcon';
import {ROUTER_LINK_COLOR, ROUTER_LINK_SIZE} from '@/components/RouterLink/constants';
import {Icons} from '@/services/icons';
import {LEFT_ICON_LINK_TEST_ID, RIGHT_ICON_LINK_TEST_ID} from '@/components/LinkWithIcon/constants';

describe('LinkWithIcon component', () => {
  afterEach(cleanup);
  const props = {
    color: ROUTER_LINK_COLOR.GRAY,
    size: ROUTER_LINK_SIZE.BOLD,
    href: "/"
  };
  test('renders component correctly', () => {
    const {container} = render(<LinkWithIcon {...props}>Link</LinkWithIcon>);
    expect(screen.getByText('Link')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  test('renders LeftIcon when provided', () => {
    const {container} = render(
      <LinkWithIcon
        LeftIcon={Icons.ArrowNarrowLeft}
        {...props}
      >
        Click
      </LinkWithIcon>,
    );
    expect(screen.getByTestId(LEFT_ICON_LINK_TEST_ID)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test('renders RightIcon when provided', () => {
    const {container} = render(
      <LinkWithIcon
        RightIcon={Icons.ArrowNarrowRight}
        {...props}
      >
        Click
      </LinkWithIcon>,
    );
    expect(screen.getByTestId(RIGHT_ICON_LINK_TEST_ID)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test('renders both LeftIcon and RightIcon when provided', () => {
    const {container} = render(
      <LinkWithIcon
        LeftIcon={Icons.ArrowNarrowLeft}
        RightIcon={Icons.ArrowNarrowRight}
        {...props}
      >
        Click
      </LinkWithIcon>,
    );
    expect(screen.getByTestId(LEFT_ICON_LINK_TEST_ID)).toBeInTheDocument();
    expect(screen.getByTestId(RIGHT_ICON_LINK_TEST_ID)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});