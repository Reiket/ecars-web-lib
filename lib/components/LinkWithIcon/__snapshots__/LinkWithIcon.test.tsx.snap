// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LinkWithIcon component > renders LeftIcon when provided 1`] = `
<div>
  <a
    class="link link--gray link--icon"
    data-testid="routerLinkTestId"
    href="/"
    rel="noopener noreferrer"
    target="_blank"
  >
    <svg
      data-testid="leftIconLinkTestId"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
      />
    </svg>
    Click
  </a>
</div>
`;

exports[`LinkWithIcon component > renders RightIcon when provided 1`] = `
<div>
  <a
    class="link link--gray link--icon"
    data-testid="routerLinkTestId"
    href="/"
    rel="noopener noreferrer"
    target="_blank"
  >
    Click
    <svg
      data-testid="rightIconLinkTestId"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
      />
    </svg>
  </a>
</div>
`;

exports[`LinkWithIcon component > renders both LeftIcon and RightIcon when provided 1`] = `
<div>
  <a
    class="link link--gray link--icon"
    data-testid="routerLinkTestId"
    href="/"
    rel="noopener noreferrer"
    target="_blank"
  >
    <svg
      data-testid="leftIconLinkTestId"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
      />
    </svg>
    Click
    <svg
      data-testid="rightIconLinkTestId"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
      />
    </svg>
  </a>
</div>
`;

exports[`LinkWithIcon component > renders component correctly 1`] = `
<div>
  <a
    class="link link--gray link--icon"
    data-testid="routerLinkTestId"
    href="/"
    rel="noopener noreferrer"
    target="_blank"
  >
    Link
  </a>
</div>
`;

exports[`LinkWithIcon component renders LeftIcon when provided 1`] = `
<div>
  <a
    class="link link--gray link--icon"
    data-testid="routerLinkTestId"
    href="/"
    rel="noopener noreferrer"
    target="_blank"
  >
    <svg
      data-testid="leftIconLinkTestId"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
      />
    </svg>
    Click
  </a>
</div>
`;

exports[`LinkWithIcon component renders RightIcon when provided 1`] = `
<div>
  <a
    class="link link--gray link--icon"
    data-testid="routerLinkTestId"
    href="/"
    rel="noopener noreferrer"
    target="_blank"
  >
    Click
    <svg
      data-testid="rightIconLinkTestId"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
      />
    </svg>
  </a>
</div>
`;

exports[`LinkWithIcon component renders both LeftIcon and RightIcon when provided 1`] = `
<div>
  <a
    class="link link--gray link--icon"
    data-testid="routerLinkTestId"
    href="/"
    rel="noopener noreferrer"
    target="_blank"
  >
    <svg
      data-testid="leftIconLinkTestId"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
      />
    </svg>
    Click
    <svg
      data-testid="rightIconLinkTestId"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
      />
    </svg>
  </a>
</div>
`;

exports[`LinkWithIcon component renders component correctly 1`] = `
<div>
  <a
    class="link link--gray link--icon"
    data-testid="routerLinkTestId"
    href="/"
    rel="noopener noreferrer"
    target="_blank"
  >
    Link
  </a>
</div>
`;
